### PR TITLE
WIP: Adding a check for detecting incorrectly set last modified time headers

### PR DESCRIPTION
**Pull request by beenanner**
Still in progress as this check relies on https://github.com/sitespeedio/pagexray/pull/16 as returning -1 will trigger failures on the check for any request that doesn't have a last modified time header.
- [ ] Check that your change/fix has corresponding unit tests
- [x] Verify that the test works by running `npm test` and test linting by `npm run lint`
### Description

http://www.cnet.com/ has a request that has a last modified time header set in the future this is incorrect against the spec and should be flagged as needing to be fixed.

Also this test should probably go into a bestpractice folder instead of performance, but https://github.com/sitespeedio/coach/blob/master/lib/index.js#L69 will need to be modified first to include multiple directories.

Also a unit test is needed before merging.

### CODE REVIEW

**Comment by soulgalore** [comment]
This is the first one with weight 0 :) Not sure it will be picked up then, because we have checks for advice with the score < 100. I think we should create a bestpractices category for HAR also, then we can have a weight and it will be picked up and shown in the bestpractice section? I think it will not work out of the box, could be we need to add something more to just get that category up and running, let me know if you want me to help you with that!

Thanks for advice, it's super cool!

**Comment by beenanner** [comment]
Sounds good @soulgalore it does look like we will need to tweak [getHarAdvice method](https://github.com/sitespeedio/coach/blob/master/lib/index.js#L69) as it only takes in account the performance folder.

File: lib/har/bestpractice/lastModifiedTime.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+module.exports = {
+  id: 'lastModifiedTime',
+  title: 'Avoid setting last modified header in the future',
+  description: 'Dont\'t set last modified headers in the future!',
```
Comment (XhmikosR) [inline lib/har/bestpractice/lastModifiedTime.js]: Small typo "Dont\'t"

File: lib/har/bestpractice/lastModifiedTime.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  weight: 0,
+  tags: ['performance', 'bestpractice', 'server'],
+
+  processPage: function(page) {
+    let score = 100;
+    let offending = [];
+    page.assets.forEach(function(asset) {
+      if (asset.timeSinceLastModified && parseInt(asset.timeSinceLastModified) < 0) {
```
Comment (XhmikosR) [inline lib/har/bestpractice/lastModifiedTime.js]: Maybe add the radix parameter too.

Comment (beenanner) [inline lib/har/bestpractice/lastModifiedTime.js]: Thanks @XhmikosR for the feedback! I've neglected this pull request for a while. I'm going to try to pick it back up after the release of 4.0 of sitespeed.

**Pull request closed without merging**